react-native-syrf-client/android/src/main/java/com/reactnativesyrfclient/SyrfClientModule.kt

```diff
@@ -96,141 +92,155 @@ class SyrfClientModule(reactContext: ReactApplicationContext) :
     permissions: Array<out String>,
     grantResults: IntArray
   ): Boolean {
-    PermissionsManager(activity).handleResults(
-      permissions,
-      successCallback = {
-        if (waitingForLocationPermission) {
-          SYRFLocation.subscribeToLocationUpdates(activity)
-        }
-        waitingForLocationPermission = false
-        if (waitingForCurrentLocationPermission) {
-          getCurrentLocation()
-        }
-        waitingForCurrentLocationPermission = false
-      },
-      exceptionCallback = {
-        waitingForLocationPermission = false
-      })
-
-    return true
+    if (currentActivity != null) {
```

Comment:
Please don't do many levels of if. It's hard to understand. Better to have the structure more flat. If (activity === null) {return false}. And so on